protocol/rpcconsumer/consumer_relay_state_machine.go

```diff
@@ -84,14 +83,24 @@ func (crsm *ConsumerRelayStateMachine) GetSelection() Selection {
 	return crsm.selection
 }
 
-func (crsm *ConsumerRelayStateMachine) ShouldRetry(numberOfRetriesLaunched int) bool {
+func (crsm *ConsumerRelayStateMachine) shouldRetryOnResult(numberOfRetriesLaunched int, numberOfNodeErrors uint64) bool {
+	shouldRetry := crsm.shouldRetryInner(numberOfRetriesLaunched)
+	// archive functionality will be added here.
+	return shouldRetry
+}
+
+func (crsm *ConsumerRelayStateMachine) shouldRetryInner(numberOfRetriesLaunched int) bool {
 	if numberOfRetriesLaunched >= MaximumNumberOfTickerRelayRetries {
 		return false
 	}
 	// best result sends to top 10 providers anyway.
 	return crsm.selection != BestResult
 }
 
+func (crsm *ConsumerRelayStateMachine) shouldRetryTicker(numberOfRetriesLaunched int) bool {
```

Comment:
this is exactly the same as shouldRetryOnResult, the change from ShouldRetry seems pointless